contracts/data_storages/RideHailingDisputesDataStorage.sol

```diff
@@ -47,6 +53,56 @@ contract RideHailingDisputesDataStorage is DataStorageBaseContract {
         return disputeData[disputeId];
     }
 
+    function getRideId(
+        uint256 disputeId
+    ) external view internalContractsOnly returns (uint) {
+        return disputeData[disputeId].rideId;
+    }
+
+    function setDisputeResponse (
+        uint256 disputeId
+    ) external internalContractsOnly {
+        disputeData[disputeId].responded = true;
+    }
+
+    function getNumOfDefedantUnresponded( //check if have respond to disputes
+        address defendant
+    ) external view internalContractsOnly returns (uint256) {
+        uint unrespondCounter = 0;
+        for(uint i = 0; i < disputeData.length; i++) {
+            if(disputeData[i].defendant == defendant && disputeData[i].responded == false)
+            unrespondCounter ++;
+        }
+        return unrespondCounter;
+    }
+
+    function getTimeRemaining(uint256 disputeId) external view internalContractsOnly returns (uint256) {
+        uint256 timePassed = block.timestamp - disputeData[disputeId].startTime;
+        if(timePassed >= 1 days) {
+            return 0;
+        } else {
+            return 1 days - timePassed;
+        }
+    }
+
+
+    function generateDisputeForVoting() external view returns (uint256) {
+        // dummy oracle : get unresolved dispute by rng
+        for (uint256 i = 0; i < disputeData.length; i++) {
+            if (disputeData[i].resolved != true) {
+                return i; // i is the disputeId
+            }
+        }
+        // returns max uint256 value if no unresolved dispute found
+        return (2 ^ (256 - 1));
```

Comment:
Hmm slightly prefer making 0 the invalid disputeId instead, i think we are doing that for Ride as well